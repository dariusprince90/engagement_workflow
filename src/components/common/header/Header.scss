@import '~@Company/web-theme-03/src/scss/variables';
@import '../../../scss/variables';

header.common-page-header {
  background-color: $pm-blue;
  height: 80px;
  overflow: hidden;

  // flex container properties
  display: flex;
  flex-direction: row;
  align-items: center;

  img.pm-logo {
    height: 50px;
    margin-left: 0.25rem;
  }

  .titles-container {
    margin-left: 1rem;
    color: $white;
    white-space: nowrap;

    // flex item properties
    flex-grow: 1;

    // flex container properties
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: center;
  }

  .page-title {
    font-size: 1.6rem;
    font-weight: 500;
    margin-bottom: 0.375rem;
  }

  .page-subtitle {
    font: normal 0.8rem $font-family-sans-serif-001;
    margin-bottom: 0.5rem;
    margin-left: 0.0625rem;
  }

  .page-help-container {
    margin-right: 0.5rem;

    // flex item properties
    flex-grow: 0;

    a {
      text-decoration: none;

      .help-logo {
        color: $pm-yellow;
        margin-right: 0.25rem;
      }

      .page-help-text {
        color: $white;
      }
    }
  }
}
