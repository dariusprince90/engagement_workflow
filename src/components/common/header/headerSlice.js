import { createSlice } from '@reduxjs/toolkit';

// **********************************************************************
// * constants

// **********************************************************************
// * functions

// **********************************************************************
// * thunks

// **********************************************************************
// * initial state

const initialState = { pageTitle: '', pageSubtitle: '' };

// **********************************************************************
// * slice

const headerSlice = createSlice({
  name: 'pageHeader',
  initialState,

  reducers: {
    pageTitleChanged: (state, action) => {
      state.pageTitle = action.payload;
    },

    pageSubtitleChanged: (state, action) => {
      state.pageSubtitle = action.payload;
    }
  },

  extraReducers: (builder) => {}
});

// **********************************************************************
// * actions

export const { pageTitleChanged, pageSubtitleChanged } = headerSlice.actions;

// **********************************************************************
// * selectors

export const selectPageTitle = (state) => state.pageHeader.pageTitle;
export const selectPageSubtitle = (state) => state.pageHeader.pageSubtitle;

// **********************************************************************
// * reducer

export default headerSlice.reducer;
