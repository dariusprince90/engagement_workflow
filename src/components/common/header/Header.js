import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import './Header.scss';
import { selectPageTitle, selectPageSubtitle } from './headerSlice';

const Header = () => {
  // **********************************************************************
  // * constants

  const pageTitle = useSelector(selectPageTitle);
  const pageSubtitle = useSelector(selectPageSubtitle);

  // **********************************************************************
  // * component vars

  // **********************************************************************
  // * functions

  // **********************************************************************
  // * side effects

  // **********************************************************************
  // * render

  return (
    <header className="common-page-header">
      {/* pm logo */}
      <Link to="/" reloadDocument>
        <img src="/images/pm-logos/pm-logo-white-small.png" alt="Company logo" className="pm-logo" />
      </Link>

      <span className="titles-container">
        {/* page title and subtitle */}
        <h3 className="page-title">{pageTitle}</h3>
        <h5 className="page-subtitle">{pageSubtitle}</h5>
      </span>

      <div className="page-help-container">
        <a
          href={
            'https://Company.sharepoint.com/sites/kshare-Internal-Accounting/Shared%20Documents/Forms/AllItems.aspx' +
            '?id=%2Fsites%2Fkshare%2DInternal%2DAccounting%2FShared%20Documents%2FNEW%20Best%20Practices%20and' +
            '%20Frequently%20Asked%20Questions%2Epdf&parent=%2Fsites%2Fkshare%2DInternal%2DAccounting%2FShared%20Documents'
          }
          target="_blank"
          rel="noopener noreferrer">
          <FontAwesomeIcon icon={['fas', 'question-circle']} className="help-logo" />
          <span className="page-help-text">Best Practices and FAQs</span>
        </a>
      </div>
    </header>
  );
};

export default Header;
