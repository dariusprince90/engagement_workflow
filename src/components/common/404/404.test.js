import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import * as reactRedux from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import faker from '@faker-js/faker';

import * as headerSlice from '../../common/header/headerSlice';
import PageNotFound404 from './404';

// **********************************************************************
// * constants

const mockDispatch = jest.fn();

const componentToRender = (
  <MemoryRouter>
    <PageNotFound404 />
  </MemoryRouter>
);

// **********************************************************************
// * mock external dependencies

jest.mock('react-redux', () => {
  return {
    useDispatch: jest.fn()
  };
});

jest.mock('../../common/header/headerSlice');

// **********************************************************************
// * unit tests

describe('PageNotFound404', () => {
  // **********************************************************************
  // * setup

  beforeAll(() => {
    jest.spyOn(headerSlice, 'pageTitleChanged');
  });

  beforeEach(() => {
    reactRedux.useDispatch.mockReturnValue(mockDispatch);
  });

  // **********************************************************************
  // * tear-down

  afterAll(() => {
    headerSlice.pageTitleChanged.mockClear();
  });

  // **********************************************************************
  // * execution

  describe('rendering', () => {
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(componentToRender, div);
    });

    it('renders correct layout', () => {
      const tree = renderer.create(componentToRender).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('functional', () => {
    it('dispatches pageTitleChanged when mounting', () => {
      // * ARRANGE
      const expectedPageTitle = 'Page Not Found';
      const results = faker.lorem.words();
      headerSlice.pageTitleChanged.mockReturnValue(results);

      // * ACT
      render(componentToRender);

      // * ASSERT
      expect(headerSlice.pageTitleChanged).toHaveBeenCalledWith(expectedPageTitle);
      expect(mockDispatch).toHaveBeenCalledWith(results);
    });
  });
});
