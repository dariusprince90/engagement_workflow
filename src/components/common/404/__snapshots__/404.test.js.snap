// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound404 rendering renders correct layout 1`] = `
<div
  className="page-not-found-404-container"
>
  <img
    alt="logo"
    className="pm-logo"
    src="/images/pm-logos/pm-logo-wide-color-small.png"
  />
  <h1
    className="display-1 mb-2"
  >
    404
  </h1>
  <h2
    className="mb-4"
  >
    Oops! You can’t make your mark here.
  </h2>
  <p>
    We cannot find the page that you’re looking for.
  </p>
  <p>
    Try the
    <a
      href="/"
      onClick={[Function]}
    >
      home page
    </a>
     instead. That is a good place to start.
  </p>
</div>
`;
